File: opencvplay/sho.cpp
#include <opencv2/core.hpp>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>
#include <iostream>

using namespace std;
using namespace cv;

int main()
{
    string image_path = samples::findFile("aero3.jpg");
    Mat img = imread(image_path, IMREAD_COLOR);
    if(img.empty()) {
        cout << "Could not read the image: " << image_path << endl;
        return 1;
    }

    imshow("Display window", img);
    int k = waitKey(0); // Wait for a keystroke in the window
    if(k == 's')
# if key pressed happened to be s, write an image.
        imwrite("aero3.png", img);

    return 0;
}

File: opencvplay/sho.py
#!/usr/bin/env python3
# this script does what?
import sys 
import cv2 as cv

argquan=len(sys.argv)
if argquan != 0:
   print("This script requires no arguments")
   sys.exit(2)

# refer to arguments with sys.argv[1] etc.
img = cv.imread(cv.samples.findFile("aero3.jpg"))
if img is None:
    sys.exit("Could not read the image.")
cv.imshow("Display window", img)
k = cv.waitKey(0)
if k == ord("s"):
    cv.imwrite("aero3.png", img)
